firefox_ui_harness/testcase.py

```diff
@@ -5,23 +5,37 @@
 from marionette import MarionetteTestCase
 
 from firefox_puppeteer import Puppeteer
+from firefox_puppeteer.ui.windows import BrowserWindow
 
 
 class FirefoxTestCase(MarionetteTestCase, Puppeteer):
-    """
-    Test case that inherits from a Puppeteer object so Firefox specific
-    libraries are exposed to test scope.
+    """Base testcase class for Firefox Desktop tests.
+
+    It enhances the Marionette testcase by inserting the Puppeteer mixin class,
+    so Firefox specific API modules are exposed to test scope.
     """
     def __init__(self, *args, **kwargs):
         MarionetteTestCase.__init__(self, *args, **kwargs)
 
+    def _init_data(self):
+        self.marionette.set_context('chrome')
+
+        self.browser = self.windows.switch_to(lambda win: type(win) is BrowserWindow)
+
+    def restart(self, flags=None):
+        """Restart Firefox and re-initialize data.
+
+        :param flags: Specific restart flags for Firefox
+        """
+        self.marionette.restart()
+        self._init_data()
```

Comment:
Ups, sorry. That is the FirefoxTestCase class. So we should always have that line here find the right window. In this case the browser window. It will not necessarily be the focused one right now. If we don't do that here, we can fail if Firefox restarts with a set of different windows open, and the `current_chrome_window_handle` is not a browser window.